src/rust/engine/src/nodes.rs

```diff
@@ -830,6 +831,47 @@ impl From<ReadLink> for NodeKey {
 }
 
 ///
+/// A Node that represents reading a file and fingerprinting its contents.
+///
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct DigestFile(File);
+
+impl Node for DigestFile {
+  type Output = fs::Digest;
+
+  fn run(self, context: Context) -> NodeFuture<fs::Digest> {
+    let file = self.0.clone();
+    context
+      .core
+      .vfs
+      .read_file(&self.0)
+      .map_err(move |e| {
+        throw(&format!(
+          "Error reading file {:?}: {}",
```

Comment:
did rustfmt force this onto separate lines?... Odd

Reply:
Yes. I continue to be pretty unimpressed with rustfmt.